shared/reducers/chat2.js

```diff
@@ -558,8 +561,14 @@ const rootReducer = (state: Types.State = initialState, action: Chat2Gen.Actions
             const messages = convoToMessages[cid]
             messages.forEach(message => {
               const m = canSendType(message)
-              const old = m ? findExisting(conversationIDKey, m) : null
-              const toSet = old ? Constants.upgradeMessage(old, message) : message
+              const oldSentOrPending = m ? findExistingSentOrPending(conversationIDKey, m) : null
```

Comment:
the logic here is getting kinda hairy -- nested ternaries, etc. It could be simplified by handling checking `m` first:

```
let toSet = message
if (m) {
  const oldSetOrPending = findExisting...
  if (oldSetOrPending) {
    toSet = Constants.upgradeMessage(oldSetToPending, ..., false)
  } else {
    const previousMessage = previousMessageMap.getIn(...)
    if (previousMessage) {
      toSet = Constants.upgradeMessage(previousMessage..., true)
    }
  }
}
```
That way you can make the type of `_old` in `upgradeMessage` non-null again.